libuv: expose 'toDisable' from nix

teto opened the pull request:
I wanted to test libuv 1.47 but this broke one of the test. Instead of copy/pasting the full list of tests to disable, this exposes it so that one can write instead:

```
          libuv_147 = prev.libuv.overrideAttrs(oa: rec {
            version =  "1.47.0";
            src = prev.fetchFromGitHub {
              owner = "libuv";
              repo = "libuv";
              rev = "v${version}";
              sha256 = "sha256-J6qvq///A/tr+/vNRVCwCc80/VHKWQTYF6Mt1I+dBCU=";
            };
            doCheck = false;

            toDisable = oa.toDisable ++ [ "tcp_connect6_link_local" ];
          });

```
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Apparently triggers ~60k rebuilds across platforms so staging it is

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

teto committed:
libuv: expose 'toDisable' from nix

I wanted to test libuv 1.47 but this broke one of the test.
Instead of copy/pasting the full list of tests to disable, this exposes
it so that one can write instead:

          libuv_147 = prev.libuv.overrideAttrs(oa: rec {
            version =  "1.47.0";
            src = prev.fetchFromGitHub {
              owner = "libuv";
              repo = "libuv";
              rev = "v${version}";
              sha256 = "sha256-J6qvq///A/tr+/vNRVCwCc80/VHKWQTYF6Mt1I+dBCU=";
            };
            doCheck = false;

            toDisable = oa.toDisable ++ [ "tcp_connect6_link_local" ];
          });

marsam commented on pkgs/development/libraries/libuv/default.nix:
```diff
@@ -35,47 +37,48 @@ stdenv.mkDerivation (finalAttrs: {
 
   outputs = [ "out" "dev" ];
 
+  toDisable = [
```
What do you think of renaming this variable `disabledTests`?

teto replied:
I fully agree. Done

Artturin commented on pkgs/development/libraries/libuv/default.nix:
```diff
@@ -26,6 +26,8 @@ stdenv.mkDerivation (finalAttrs: {
   version = "1.46.0";
   pname = "libuv";
 
+  __structuredAttrs = true;
```
The package has to be checked and diffed on all supported OS'es when enabling structuredAttrs because there may be hooks and platform specific hooks which don't work with it.

Alternative is to put `disabledTests` in to `passthru`

teto replied:
thanks for the precision but how is a simple contributor like myself supposed to check that ? is there some kind of tooling or anyone to ping who can confirm it works on all platforms ?

Artturin replied:
I don't know, im hesitant to set `__structuredAttrs` because only the core kinda works with it.

teto replied:
it's sad we can't use __structuredAttrs after all these years. passthru would be fine here but I would prefer if we could enable __structuredAttrs. So maybe let's wait for after the release and then:
- see if people with the knowhow can actually confirm it's ok
- alternatively merge and fix whatever needs fixing or revert if too painful 

I've enabled _structuredAttrs in the lua ecosystem without anyone complaining so I am tempted to think __structuredAttrs  could work here too without problem.

Artturin replied:
There's also the point that this list doesn't or shouldn't be a part of the drv since it's referred to directly via finalAttrs

Bot-wxt1221 committed:
calligraplan: init at 3.3.0

qbisi committed:
petsc: simplify configurePhase

qbisi committed:
petsc: reformat with nixfmt-rfc-style

qbisi committed:
python312Packages.petsc4py: init at 3.22.3

qbisi committed:
petsc: add standalone metis support

qbisi committed:
petsc: add ptscotch support

qbisi committed:
petsc: add scalapack and mumps support

qbisi committed:
petsc: add withFullDeps option

qbisi committed:
petsc: rename some optional dependency flags

qbisi committed:
petsc: use lib.{getDev,getLib} instead *.{dev,lib}

kilianar committed:
calibre: 7.24.0 -> 7.26.0

https://github.com/kovidgoyal/calibre/releases/tag/v7.26.0

Bot-wxt1221 committed:
greed: init at 4.3

aduh95 committed:
nodejs_23: 23.7.0 -> 23.8.0

TimQuelch committed:
bash-my-aws: 2020-01-11 -> 2025-01-22 and refactor

Pull to latest commit and refactor the derivation.

qbisi committed:
petsc: reorder build options and external dependency

qbisi committed:
petsc: add hdf5 fortran binding

qbisi committed:
petsc: set optimized and withHdf5 default to true

Petsc is commonly optimized built with hdf5 support to be used
as a component of other software and project.
See
    1. https://sources.debian.org/src/petsc/3.22.2%2Bdfsg1-1/debian/rules/#L154
    2. https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=petsc

aos committed:
python3Packages.llm-anthropic: init at 0.12

Closes #327800
Closes #372713

(Note: This is just #373495 upgraded to `0.12`. Seems like something
happened with git/Github and my PR was automatically closed.)

This is a plugin for the already packaged `llm` software. There exists 2 PRs related to this:
- #327800: the original
- #372713: forks the original and updates as it had no activity for multiple months

As helpfully pointed out by @ShaddyDC (https://github.com/NixOS/nixpkgs/pull/372713#issuecomment-2586795561) this is now superseded by `llm-anthropic`, of which this is meant to package.

(We also have other PRs that have been commited that add similr plugins at #359379 and #344746.)

markuskowa committed:
gromacs: 2024.5 -> 2025.0

markuskowa committed:
gromacs: apply nixfmt

imincik committed:
python3Packages.owslib: 3.32.0 -> 3.32.1

r-ryantm committed:
qownnotes: 25.2.1 -> 25.2.7

r-ryantm committed:
snpguest: 0.8.0 -> 0.8.3

Gliczy committed:
vencord: 1.11.4 -> 1.11.5

r-ryantm committed:
tparse: 0.16.0 -> 0.17.0

xiaoxiangmoe committed:
ast-grep: 0.34.4 -> 0.35.0

r-ryantm committed:
typos: 1.29.4 -> 1.29.7

nixpkgs-merge-bot[bot] committed:
qownnotes: 25.2.1 -> 25.2.7 (#382670)

Co-authored-by: pbek <patrizio@bekerle.com>

nixpkgs-merge-bot[bot] committed:
typos: 1.29.4 -> 1.29.7 (#382743)

Co-authored-by: mgttlinger <mgttlinger@users.noreply.github.com>

xiaoxiangmoe committed:
windsurf: 1.2.6 -> 1.3.4

GaetanLepage committed:
calligraplan: init at 3.3.0 (#376426)

GaetanLepage committed:
ast-grep: 0.34.4 -> 0.35.0 (#382741)

GaetanLepage committed:
greed: init at 4.3 (#381720)

drupol committed:
windsurf: 1.2.6 -> 1.3.4 (#382758)

r-ryantm committed:
jjui: 0.3 -> 0.4

minijackson committed:
python3Packages.llm-anthropic: init at 0.12 (#382108)

minijackson committed:
calibre: 7.24.0 -> 7.26.0 (#380718)

r-ryantm committed:
mago: 0.0.18 -> 0.10.0

MisileLab committed:
pnpm: 10.4.0 -> 10.4.1

Signed-off-by: misilelab <misileminecord@gmail.com>

tomberek committed:
bash-my-aws: 2020-01-11 -> 2025-01-22 and refactor (#381934)

nixpkgs-merge-bot[bot] committed:
mago: 0.0.18 -> 0.10.0 (#382779)

Co-authored-by: gaelreyrol <me@gaelreyrol.dev>

imincik committed:
python3Packages.owslib: 3.32.0 -> 3.32.1 (#382661)

Scrumplex committed:
vencord: 1.11.4 -> 1.11.5 (#382727)

Scrumplex committed:
pnpm: 10.4.0 -> 10.4.1 (#382778)

katexochen committed:
snpguest: 0.8.0 -> 0.8.3 (#382723)

nixpkgs-merge-bot[bot] committed:
jjui: 0.3 -> 0.4 (#382769)

Co-authored-by: Adda0 <chocholaty.david@protonmail.com>

dasJ committed:
nodejs_23: 23.7.0 -> 23.8.0 (#381837)

nixpkgs-merge-bot[bot] committed:
tparse: 0.16.0 -> 0.17.0 (#382734)

Co-authored-by: obreitwi <obreitwi@users.noreply.github.com>

markuskowa committed:
gromacs: 2024.5 -> 2025.0 (#382385)

markuskowa committed:
petsc: refactor and add petsc4py support (#379406)
